internal/exec/shell_utils.go

```diff
@@ -334,3 +346,34 @@ func execTerraformShellCommand(
 	u.LogDebug(atmosConfig, fmt.Sprintf("Exited shell: %s\n", state.String()))
 	return nil
 }
+
+// mergeEnvVars combines two sets of environment variables, with overrideEnv taking precedence.
+//
+// This is necessary because:
+// 1. We need to preserve existing system environment variables (PATH, HOME, etc.)
+// 2. Component-specific variables (TF_CLI_ARGS, ATMOS_* vars) must take precedence
+// 3. (Most importantly) Simply appending the two sets wouldn't work as processes use the first occurrence of each variable
```

Comment:
Can you say this in a different way? 

> wouldn't work as processes use the first occurrence of each variable

That's only desirable for TF_ vars. But not the general solution.

Reply:
> When you append `os.Environ()` and `componentEnvList`, if both contain the same environment variable (KEY), the last occurrence in the list takes precedence.

This is generally the correct behavior. The _exception_ is with TF_ vars that atmos modifies and there's a canonical way to add parameters. This is not true for all other ENVs.

Reply:
```
// mergeEnvVars combines two sets of environment variables, with overrideEnv taking precedence.
//
// This is necessary because:
//  1. We need to preserve existing system environment variables (PATH, HOME, etc.)
//  2. Atmos-specific variables (TF_CLI_ARGS, ATMOS_* vars) must take precedence
//  3. For conflicts, such as TF_CLI_ARGS_*, we need special handling to ensure proper merging rather than simple overwriting
```